static/app/components/notificationActions/notificationActionManager.tsx

```diff
@@ -0,0 +1,203 @@
+
+  // Groups the notification actions together by service
+  // Will render the notif actions in the order the keys are listed in
+  const actionsMap: Record<
+    NotificationActionService,
+    {action: NotificationAction; index: number}[]
+  > = {
+    [NotificationActionService.SENTRY_NOTIFICATION]: [],
+    [NotificationActionService.EMAIL]: [],
+    [NotificationActionService.SLACK]: [],
+    [NotificationActionService.PAGERDUTY]: [],
+    [NotificationActionService.MSTEAMS]: [],
+    [NotificationActionService.SENTRY_APP]: [],
+  };
+  notificationActions.forEach((action, index) => {
+    if (action.serviceType) {
+      actionsMap[action.serviceType].push({action, index});
+    }
+  });
+
+  // Groups the pagerduty integrations with their corresponding allowed services
+  const pagerdutyIntegrations: Record<number, AvailableNotificationAction[]> = {};
+  availableServices[NotificationActionService.PAGERDUTY].forEach(service => {
+    const integrationId = service.action.integrationId;
+    if (integrationId) {
+      if (integrationId in pagerdutyIntegrations) {
+        pagerdutyIntegrations[integrationId].push(service);
+      } else {
+        pagerdutyIntegrations[integrationId] = [service];
+      }
+    }
+  });
+
+  function renderNotificationActions() {
+    if (!notificationActions) {
+      return null;
+    }
+
+    const serviceActions: NotificationAction[] = [];
+
+    // Renders the notif actions grouped together by kind
+    Object.keys(actionsMap).forEach(serviceType => {
+      const services = actionsMap[serviceType];
+      serviceActions.push(
+        services.map(({action, index}) => (
+          <NotificationActionItem
+            key={index}
+            index={index}
+            defaultEdit={!action.id}
+            action={action}
+            recipientRoles={recipientRoles}
+            availableActions={availableServices[serviceType]}
+            pagerdutyIntegrations={pagerdutyIntegrations}
+            project={project}
+            onDelete={removeNotificationAction}
+            onUpdate={updateNotificationAction}
+          />
+        ))
+      );
+    });
+    return serviceActions;
+  }
+
+  const getMenuItems = () => {
+    const menuItems: MenuItemProps[] = [];
+    Object.entries(availableServices).forEach(([serviceType, validActions]) => {
+      if (validActions.length === 0) {
+        return;
+      }
+      // Cannot have more than one Sentry notification
+      if (
+        serviceType === NotificationActionService.SENTRY_NOTIFICATION &&
+        actionsMap[serviceType].length === 1
+      ) {
+        return;
+      }
+
+      menuItems.push({
+        key: serviceType,
+        label: t(
+          'Send a %s notification',
+          serviceType !== NotificationActionService.SENTRY_NOTIFICATION
+            ? capitalize(serviceType)
+            : 'Sentry'
+        ),
+        onAction: () => addNotificationAction(validActions[0].action),
+      });
+    });
+    return menuItems;
```

Comment:
same comment - return the `.map`

Reply:
`.map` is not possible on an object?

Reply:
Oh.

I think i'm referring to 
```
Object.entries(availableServices).forEach
```

we're iterating through the entries, then populating `menuItems`
Then once we've finished iterating, we return `menuItems`

may as well just return a map

Reply:
ah i can't do `.map` on this because for some conditions, i don't render anything so the map expecting the same number of outputs as inputs would be broken